misaligned.c

```diff
@@ -1,21 +1,56 @@
 #include <stdio.h>
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
+typedef struct
+{
+    int pairNumber;
+    const char * majorColor;
+    const char * minorColor;
+} colorMapStruct;
+
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+colorMapStruct ColorMaps[30];
+
+
+int GetColorMap(colorMapStruct * colorMaps)
+{
+    int i = 0, j = 0, index = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            colorMaps[index].pairNumber = i * 5 + j;

```

Comment:
Magic number 5 ! what happens when number of minor colors is not 5 and something else? will this formula hold good?
Do we have enough tests to catch this type of failure?

Reply:
whole functionality is retained as-is from its predecessor. As we were instructed not to correct anything from this code. Hence moved this functionality to separate function, so, as the enable a possibility to test this functionality. Do you suggest to change the main code now itself @prithamvshetty

Reply:
Production code modification not important for this session. Adding test cases to define the behaviour of production code is important 😊